<launch>
  <!-- Camera Nodes using gscam for Jetson Nano -->
  <group ns="stereo">
    <!-- Left Camera -->
    <node pkg="robot_main" type="single_camera.py" name="left_camera" output="screen">
      <param name="camera_id" value="0"/> <!-- CSI camera 0 -->
      <param name="width" value="640"/>
      <param name="height" value="480"/>
      <param name="fps" value="20"/>
      <param name="camera_frame_id" value="camera_left_link"/>
      <remap from="image_raw" to="/stereo/left/image_raw"/>
      <remap from="camera_info" to="/stereo/left/camera_info"/>
      <remap from="set_camera_info" to="/stereo/left/set_camera_info"/>
      <param name="camera_info_url" value="file://$(find robot_main)/config/left_camera.yaml"/>
    </node>
    
    <!-- Right Camera -->
    <node pkg="robot_main" type="single_camera.py" name="right_camera" output="screen">
      <param name="camera_id" value="1"/> <!-- CSI camera 1 -->
      <param name="width" value="640"/>
      <param name="height" value="480"/>
      <param name="fps" value="20"/>
      <param name="camera_frame_id" value="camera_right_link"/>
      <remap from="image_raw" to="/stereo/right/image_raw"/>
      <remap from="camera_info" to="/stereo/right/camera_info"/>
      <remap from="set_camera_info" to="/stereo/right/set_camera_info"/>
      <param name="camera_info_url" value="file://$(find robot_main)/config/right_camera.yaml"/>
    </node>
    
    <!-- Custom Stereo Processor -->
    <node pkg="robot_main" type="stereo_processor_c" name="stereo_processor" output="screen"/>
    
    <!-- TF for camera frames -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="left_camera_frame"
          args="0 0 0 0 0 0 base_link left_camera_frame 100"/>
    <node pkg="tf" type="static_transform_publisher" name="right_camera_frame"
          args="0 -0.065 0 0 0 0 base_link right_camera_frame 100"/> -->
    
    <!-- Visualization -->
    <!-- <node pkg="rviz" type="rviz" name="rviz" 
          args="-d $(find robot_main)/config/stereo_cloud.rviz" required="false"/> -->
  </group>
</launch>
